### ISSUE TITLE
No custom auth drivers work with 3.x

### ISSUE THREAD

**Issue by tm1000** [WeblateOrg/weblate#2131]
In #1883 it's mentioned that custom Django auth drivers work with Weblate however since 3.x this is no longer true because of the massive rewrite to User and Group models in Weblate 3

This worked previously in 2.x

using django-crowd-auth one of the first errors I encounted was this:
```
Request Method: | POST
-- | --
https://{site}/accounts/login/
2.0.7
TypeError
'is_staff' is an invalid keyword argument for this function
/usr/local/lib/python3.5/dist-packages/django/db/models/base.py in __init__, line 495
/usr/bin/uwsgi-core
3.5.3
['/usr/local/lib/python3.5/dist-packages/',  '.',  '',  '/usr/lib/python35.zip',  '/usr/lib/python3.5',  '/usr/lib/python3.5/plat-x86_64-linux-gnu',  '/usr/lib/python3.5/lib-dynload',  '/usr/local/lib/python3.5/dist-packages',  '/usr/lib/python3/dist-packages']
Thu, 19 Jul 2018 05:53:48 +0000
```

This is because if you review https://github.com/WeblateOrg/weblate/blob/master/weblate/auth/models.py#L253 it's missing in the abstract.

The same applies to first_name and last_name

If I simplify the User() call to only call username,full_name,email,is_superuser,is_active it will work on creating the user but then when it calls user.groups.set(groups) later it fails with

```
'Group' instance expected, got <Group: All Office List>
```

https://github.com/pmuller/django-crowd-auth

**Comment by tm1000** [comment]
I fixed this for django-crowd-auth in https://github.com/tm1000/django-crowd-auth/commit/0c2a23af19407ad44747ae139af40d1e703a6a3a

However now it's a fork of the real auth and it only works with weblate. Unless I'm missing something all auth plugins will have to work this way (I think I am missing something though)

**Comment by nijel** [comment]
The `get_user_model` should work just fine with Weblate. We don't use `django.contrib.auth.models.Group` at all, not sure if there is a good way to address this. As for `is_staff`, it should be pretty easy to fix on Weblate side.

**Comment by nijel** [comment]
...I've hopefully addressed `is_staff` in f7a931120.

**Comment by tm1000** [comment]
@nijel yes get_user_model did work besides the issue with is_staff :-) 

I don't know what to do about ```django.contrib.auth.models.Group``` either. I don't know if it's common for third party authentication drivers to modify Groups in Django. I realize the error ```'Group' instance expected, got <Group: All Office List>``` is basically saying the Group "All Office List" is an instance of Django Group and not weblate.group. So any external auth driver that wants to modify groups will have to import the weblate.group and not django group.

The new group and role methodology is much better in weblate 3.x so it's a decent trade off I think and again I'm not sure how common it is for external authentication addons to be modifying groups.

**Comment by tm1000** [comment]
Ah it also tripped up on first_name and last_name though I see in save you have a work around so I am not sure where it didn't work

**Comment by nijel** [comment]
Did you get some error on first/last_name? Or it just was not saved?

Also can you please post traceback for the 'Group' instance expected error? I think I have idea how to workaround that as well...

The User model is interchangable in Django, but Group one is not. Unfortunately sticking with Django Group model would make things too complex, so we've decided to replace that.
